### PR TITLE
Faster precommit

### LINKED ISSUE #379: Precommit script takes too long
## Describe the bug
Pylint takes a little too long. See if we can make `pylint`, `eslint`, and Django tests run faster.

## To Reproduce
Steps to reproduce the behavior. For example:
1. Run `./precommit` in the root directory.

## Expected behavior
No specific benchmark, but preferably it should be as fast as possible. 

## Screenshots
None

## Your Environment:
 - OS: WSL2 (Ubuntu 20.04) on Windows 10

## Additional context
From @j-alicia-long:
> [Is it just me or does pylint take a long time (>1 min) to run?](https://github.com/thecourseforum/theCourseForum2/pull/372#pullrequestreview-585032222)

### PR DESCRIPTION
**Pull request by Bartleby2718**
## Status
- Done

## GitHub Issues addressed (`#IssueNumber`)
- This PR closes #379.

## What I did
- Added more options commands in `precommit` for faster tests/linting

## Tests
- Added `date "+%Y-%m-%d %H:%M:%S"` before/after each command and ran `./precommit` three times each to get the following result:
#### Before the changes:
- 1st run:
```
2021-02-13 15:50:24
2021-02-13 15:51:22
2021-02-13 15:53:18
2021-02-13 15:54:32
2021-02-13 15:54:53
```
- 2nd run:
```
2021-02-13 15:54:59
2021-02-13 15:55:49
2021-02-13 15:57:49
2021-02-13 15:58:37
2021-02-13 15:58:59
```
- 3rd run:
```
2021-02-13 15:58:59
2021-02-13 15:59:50
2021-02-13 16:01:38
2021-02-13 16:02:26
2021-02-13 16:02:48
```

#### After the changes:
- 1st run:
```
2021-02-13 16:19:31
2021-02-13 16:20:04
2021-02-13 16:20:32
2021-02-13 16:21:22
2021-02-13 16:21:33
```
- 2nd run:
```
2021-02-13 16:21:33
2021-02-13 16:22:02
2021-02-13 16:22:29
2021-02-13 16:23:21
2021-02-13 16:23:31
```
- 3rd run:
```
2021-02-13 16:23:31
2021-02-13 16:24:02
2021-02-13 16:24:30
2021-02-13 16:25:19
2021-02-13 16:25:29
```

#### Comparison:
- Before:

|  name   | 1st  | 2nd  | 3rd  | median |  mean  |
|  :---:  |:---: |:---: |:---: | :---:  | :---:  |
|autopep8 | 58s  | 50s  | 51s  |   51s  |  53s   |
| pylint  | 116s | 120s | 108s |  116s  | 114.7s | 
| eslint  | 74s  | 48s  | 48s  |  48s   | 56.7s  | 
| django  | 21s  | 22s  | 22s  |  22s   | 21.7s  | 
- After:

|  name   | 1st  | 2nd  | 3rd  | median |  mean  |
|  :---:  |:---: |:---: |:---: | :---:  | :---:  |
|autopep8 | 33s  | 29s  | 29s  |  29s   | 30.3s  |
| pylint  | 28s  | 27s  | 28s  |  28s   | 27.7s  |
| eslint  | 50s  | 52s  | 49s  |  50s   | 50.3s  |
| django  | 11s  | 10s  | 10s  |  10s   | 10.3s  |

## Questions/Discussions/Notes
- ESLint doesn't seem run any faster with `--cache`. Should I take it out?
- A note on `--keepdb` for Django tests ([Source](https://stackoverflow.com/questions/31288353/django-testing-does-keepdb-reset-changes-made-during-tests)):
> So, any data that is somehow left in the database when the tests finish will be seen as initial data. This is mostly relevant if some error or a user interrupt prevents tests without transactions from cleaning up the database. In that case it may be a good idea to recreate the database.

## Merging the PR
- Who should merge this PR?
  - [ ] I will merge it myself
  - [x] The last reviewer to approve can merge it
- Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
  - [ ] preserve the history
  - [x] squash-merge

### CODE REVIEW

**Comment by Bartleby2718** [comment]
Note: I took out `--parallel` from `manage.py test` because it interferes with calculating Django code coverage. This requires ~6.5 more seconds to run Django tests, but I have not updated the stats in the original description.

**Review by alex-shen1 — APPROVED** [review]
Looks good to me! Regarding `--cache`, I believe that the fact that we don't have too much of a speedup is because we don't have that many JS files in the first place. I'm not sure that we'll ever have _that_ much JS, but on the other hand, `.eslintcache` is gitignored so would it persist across branches? That might be problematic, so unless it works differently I'd side with removing it. Ultimately up to you though, I don't think it'll matter _too_ much either way.

From my (albeit brief) investigation, `--keepdb` is probably okay to keep. I tried interrupting the tests at random intervals and it didn't seem to have any negative effect, and our Django tests don't seem to take that long so it's unlikely that anyone not deliberately trying to break it would interrupt in the middle of a run anyways.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ jobs:
           envsubst < .config/.env.example > .env
           python manage.py makemigrations
           python manage.py migrate
-          coverage run manage.py test
+          coverage run manage.py test --keepdb
```
Comment (alex-shen1) [inline .github/workflows/continuous-integration.yml:90]: I wonder if this is necessary though? Because the CI rebuilds every time.

Comment (Bartleby2718) [reply .github/workflows/continuous-integration.yml]: Yeah, I think you're right. Removed in 72acae31ee82cac8f81401eb6ed927911e619134.

**Comment by Bartleby2718** [comment]
@alex-shen1 From what I googled before, it takes a lot of time to parse the files (before actually linting them). Maybe the cache _was_ used but it didn't matter because it was already fast. I prefer to minimize unnecessary diffs, so I took out the unnecessary `--keepdb` in `.github/workflows/continuous-integration.yml`.

**Review by alex-shen1 — APPROVED** [review]
rubberstamp go brrrr

**Comment by Bartleby2718** [comment]
Oops I thought I took out the `--cache` thing... 🤔

**Pull request merged**